platform/darwin/src/MGLCircleStyleLayer.mm

```diff
@@ -0,0 +1,79 @@
+// This file is generated. 
+// Edit platform/ios/scripts/generate-style-code.js, then run `make style-code-darwin`.
+
+#import "MGLStyleLayer_Private.hpp"
+#import "MGLCircleStyleLayer.h"
+#import <mbgl/style/layers/circle_layer.hpp>
+
+@interface MGLCircleStyleLayer() {
+    mbgl::style::CircleLayer *circleLayer;
+}
+@end
+
+@implementation MGLCircleStyleLayer
+
+- (void)setLayer:(mbgl::style::Layer *)layer
+{
+    circleLayer = reinterpret_cast<mbgl::style::CircleLayer *>(layer);
+}
+
+#pragma mark - Accessing the Paint Attributes
+
+- (void)setCircleRadius:(CGFloat)circleRadius
+{
+    circleLayer->setCircleRadius(circleRadius);
+}
+- (CGFloat)circleRadius
+{
+    return circleLayer->getCircleRadius().asConstant();
+}
+- (void)setCircleColor:(MGLColor*)circleColor
+{
+    circleLayer->setCircleColor(circleColor.mbgl_color);
+}
+- (MGLColor *)circleColor
+{
+    return [MGLColor mbgl_color:circleLayer->getCircleColor().asConstant()];
+}
+- (void)setCircleBlur:(CGFloat)circleBlur
+{
+    circleLayer->setCircleBlur(circleBlur);
```

Comment:
There’s a lot of repetition in these implementation files. Even though it’s all generated, it could potentially make the generated code easier to reason about if we factor the boilerplate code out into macros, so that each pair of `CGFloat`-typed getters and setters is ultimately represented by:

```c
DEFINE_LAYER_PROP_NUMBER(circle, CircleBlur, CircleBlur)
```

These macros would be a little tricker to debug, but I’d expect them to be pretty straightforward callthroughs to other code anyways.